### PR TITLE
Add NatsBroker.new_inbox()

### LINKED ISSUE #1542: Feature: add support for `new_inbox()` in NatsBroker
**Is your feature request related to a problem? Please describe.**

Using NATS, I'd like to establish a private communication channel between a server and a client over which multiple messages can be sent. The `nats-py` module has a [new_inbox](https://nats-io.github.io/nats.py/modules.html#nats.aio.client.Client.new_inbox) method, but this isn't currently directly supported in FastStream.

**Describe the solution you'd like**

Add a new method `NatsBroker.new_inbox()` which calls `nats.aio.client.Client.new_inbox()` under the hood.

**Feature code example**

```python
from faststream.nats import NatsBroker

broker = NatsBroker(inbox_prefix="_CUSTOM_INBOX")
await broker.connect()
new_inbox_name = await broker.new_inbox()
```
Example output:
```python
>>> print(new_inbox_name)
'_CUSTOM_INBOX.f9hUpVb8xoVAtGpJid2VLA'
```

### PR DESCRIPTION
**Pull request by maxalbert**
Fixes #1542 

## Type of change

Please delete options that are not relevant.

- [X] Documentation (typos, code examples, or any documentation updates)
- [X] New feature (a non-breaking change that adds functionality)

## Checklist

- [X] My code adheres to the style guidelines of this project (`scripts/lint.sh` shows no errors)
- [X] I have conducted a self-review of my own code
- [X] I have made the necessary changes to the documentation
- [X] My changes do not generate any new warnings
- [X] I have added tests to validate the effectiveness of my fix or the functionality of my new feature
- [X] Both new and existing unit tests pass successfully on my local environment by running `scripts/test-cov.sh`
- [X] I have ensured that static analysis tests are passing by running `scripts/static-analysis.sh`

(Note: regarding the last item, running `mypy` produced 24 errors but they are all related to the confluent broker and not to my changes.)

### CODE REVIEW

**Review by Lancetnik — APPROVED** [review]

File: faststream/nats/broker/broker.py [hunk 1]
```diff
@@ -901,3 +901,12 @@ async def wrapper() -> None:
...
         return wrapper
+
+    async def new_inbox(self) -> str:
+        """Return a unique inbox that can be used for NATS requests or subscriptions.
+
+        Calls `nats.aio.client.Client.new_inbox` [1] under the hood.
+
+        [1] https://nats-io.github.io/nats.py/modules.html#nats.aio.client.Client.new_inbox
+        """
+        return self._connection.new_inbox()
```
Comment (Lancetnik) [inline faststream/nats/broker/broker.py:912]: You should add assert here to check, that the broker is already connected (and ignore this asset by noseq)

Comment (maxalbert) [reply faststream/nats/broker/broker.py:915]: Thanks for the hint. I only saw this comment after I added a check which raises a RuntimeError. Please let me know if you'd like me to use a regular assert statement instead.

Comment (Lancetnik) [reply faststream/nats/broker/broker.py:915]: Please, use assert in this case. I prefer to use switchable checks for development.

Comment (maxalbert) [reply faststream/nats/broker/broker.py:914]: Gotcha, done. Out of curiosity, what do you mean by using "switchable checks for development"?

**Review by Lancetnik — APPROVED** [review]

**Comment by maxalbert** [comment]
FYI, I added one more line to the docstring (to mention NatsBroker's `inbox_prefix` argument) and merged the latest changes from main to bring the branch up to date.

**Review by Lancetnik — APPROVED** [review]

**Pull request merged**